src/geometry.rs

```diff
@@ -864,6 +3088,71 @@ mod tests {
         assert!(geometry_without_orphans.has_no_orphan_normals());
     }
 
+    #[test]
+    fn test_geometry_unoriented_edges_iter() {
+        let (faces, vertices, normals) = quad_with_normals();
+        let geometry = Geometry::from_triangle_faces_with_vertices_and_normals(
+            faces.clone(),
+            vertices.clone(),
+            normals.clone(),
+        );
+        let unoriented_edges_correct = vec![
+            UnorientedEdge(OrientedEdge::new(0, 1)),
+            UnorientedEdge(OrientedEdge::new(1, 2)),
+            UnorientedEdge(OrientedEdge::new(2, 0)),
+            UnorientedEdge(OrientedEdge::new(2, 3)),
+            UnorientedEdge(OrientedEdge::new(3, 0)),
+            UnorientedEdge(OrientedEdge::new(0, 2)),
+        ];
+        let unoriented_edges_to_check: Vec<UnorientedEdge> =
+            geometry.unoriented_edges_iter().collect();
+
+        assert!(unoriented_edges_to_check
+            .iter()
+            .all(|u_e| unoriented_edges_correct.iter().any(|e| e == u_e)));
+
+        let len_1 = unoriented_edges_to_check.len();
+        let len_2 = unoriented_edges_correct.len();
+        assert_eq!(
+            len_1, len_2,
+            "unoriented_edges_to_check.len() = {}, unoriented_edges_correct.len() = {}",
+            len_1, len_2
+        );
+    }
+
+    #[test]
+    fn test_geometry_oriented_edges_iter() {
+        let (faces, vertices, normals) = quad_with_normals();
+        let geometry = Geometry::from_triangle_faces_with_vertices_and_normals(
+            faces.clone(),
+            vertices.clone(),
+            normals.clone(),
+        );
+
+        let oriented_edges_correct = vec![
+            OrientedEdge::new(0, 1),
+            OrientedEdge::new(1, 2),
+            OrientedEdge::new(2, 0),
+            OrientedEdge::new(2, 3),
+            OrientedEdge::new(3, 0),
+            OrientedEdge::new(0, 2),
+        ];
+        let oriented_edges_to_check: Vec<OrientedEdge> = geometry.oriented_edges_iter().collect();
+
+        assert!(oriented_edges_to_check
```

Comment:
Same as above: you need to swap the inner and outer loops for this test to actually test anything.